connect-inject/endpoints_controller_test.go

```diff
@@ -319,6 +343,29 @@ func TestProcessUpstreams(t *testing.T) {
 					LocalBindPort:   2234,
 				},
 			},
+			consulNamespacesEnabled: false,
+		},
+		{
+			name: "multiple upstreams with consul namespaces",
+			pod: func() *corev1.Pod {
+				pod1 := createPod("pod1", "1.2.3.4", true)
+				pod1.Annotations[annotationUpstreams] = "upstream1:1234, upstream2.bar:2234"
+				return pod1
+			},
+			expected: []api.Upstream{
+				{
+					DestinationType: api.UpstreamDestTypeService,
+					DestinationName: "upstream1",
+					LocalBindPort:   1234,
+				},
+				{
+					DestinationType:      api.UpstreamDestTypeService,
+					DestinationName:      "upstream2",
+					DestinationNamespace: "bar",
+					LocalBindPort:        2234,
+				},
+			},
+			consulNamespacesEnabled: true,
```

Comment:
should we add another test for upstreams with ns and datacenter to check that everything is parsed correctly?